tests/model/eventsources/test_self_managed_kafka_event_source.py

```diff
@@ -0,0 +1,297 @@
+        ]
+        self.kafka_event_source.KafkaBootstrapServers = ["endpoint1", "endpoint2"]
+        self.kafka_event_source.Enabled = True
+        self.kafka_event_source.BatchSize = 1
+
+        with self.assertRaises(InvalidEventException):
+            self.kafka_event_source.get_policy_statements()
+
+    def test_must_raise_for_empty_topics(self):
+        self.kafka_event_source.SourceAccessConfigurations = [
+            {"Type": "SASL_SCRAM_256_AUTH", "URI": "SECRET_URI"},
+            {"Type": "VPC_SUBNET", "URI": "SECRET_URI"},
+            {"Type": "VPC_SECURITY_GROUP", "URI": "SECRET_URI"},
+        ]
+        self.kafka_event_source.KafkaBootstrapServers = ["endpoint1", "endpoint2"]
+        self.kafka_event_source.Enabled = True
+        self.kafka_event_source.BatchSize = 1
+        self.kafka_event_source.Topics = []
+
+        with self.assertRaises(InvalidEventException):
+            self.kafka_event_source.get_policy_statements()
+
+    def test_must_raise_for_multiple_topics(self):
+        self.kafka_event_source.SourceAccessConfigurations = [
+            {"Type": "SASL_SCRAM_256_AUTH", "URI": "SECRET_URI"},
+            {"Type": "VPC_SUBNET", "URI": "SECRET_URI"},
+            {"Type": "VPC_SECURITY_GROUP", "URI": "SECRET_URI"},
+        ]
+        self.kafka_event_source.KafkaBootstrapServers = ["endpoint1", "endpoint2"]
+        self.kafka_event_source.Topics = ["Topics1", "Topics2"]
+        self.kafka_event_source.Enabled = True
+        self.kafka_event_source.BatchSize = 1
+
+        with self.assertRaises(InvalidEventException):
+            self.kafka_event_source.get_policy_statements()
+
+    def test_must_raise_for_missing_endpoints(self):
+        self.kafka_event_source.SourceAccessConfigurations = [
+            {"Type": "SASL_SCRAM_256_AUTH", "URI": "SECRET_URI"},
+            {"Type": "VPC_SUBNET", "URI": "SECRET_URI"},
+            {"Type": "VPC_SECURITY_GROUP", "URI": "SECRET_URI"},
+        ]
+        self.kafka_event_source.Enabled = True
+        self.kafka_event_source.Topics = ["Topics"]
+        self.kafka_event_source.BatchSize = 1
+
+        with self.assertRaises(InvalidEventException):
+            self.kafka_event_source.get_policy_statements()
+
+    def test_must_raise_for_empty_bootstrap_server(self):
+        self.kafka_event_source.SourceAccessConfigurations = [
+            {"Type": "SASL_SCRAM_256_AUTH", "URI": "SECRET_URI"},
+            {"Type": "VPC_SUBNET", "URI": "SECRET_URI"},
+            {"Type": "VPC_SECURITY_GROUP", "URI": "SECRET_URI"},
+        ]
+        self.kafka_event_source.KafkaBootstrapServers = []
+        self.kafka_event_source.Enabled = True
+        self.kafka_event_source.Topics = ["Topics"]
+        self.kafka_event_source.BatchSize = 1
+
+        with self.assertRaises(InvalidEventException):
+            self.kafka_event_source.get_policy_statements()
+
+    def test_must_raise_for_missing_vpc_security_group(self):
```

Comment:
You may have mixed the names of the two tests here.
I think this one is meant to be `test_must_raise_for_missing_vpc_subnet` as it's missing `VPC_SUBNET`.